code/modules/mob/dead/new_player/poll.dm

```diff
@@ -1,620 +1,510 @@
+			output += " checked"
+		output += ">[option.text]</label><br>"
+	output += "</td></tr></table>"
+	if(!length(voted_for) || poll.allow_revoting)
+		output += "<p><input type='submit' value='Vote'></form>"
+	output += "</div>"
+	src << browse(jointext(output, ""),"window=playerpoll;size=500x300")
 
-				</script>
-				</head>
-				<body>
-				<div align='center'><B>Player poll</B><hr>
-				<b>Question: [pollquestion]</b><br>Please sort the options in the order of <b>most preferred</b> to <b>least preferred</b><br>
-				<font size='2'>Revoting has been enabled on this poll, if you think you made a mistake, simply revote<br></font>
-				<font size='2'>Poll runs from <b>[pollstarttime]</b> until <b>[pollendtime]</b></font><p>
-				</div>
-				<form name='cardcomp' action='?src=[REF(src)]' method='POST'>
-				<input type='hidden' name='src' value='[REF(src)]'>
-				<input type='hidden' name='votepollid' value='[pollid]'>
-				<input type='hidden' name='votetype' value=[POLLTYPE_IRV]>
-				<input type='hidden' name='IRVdata' id='IRVdata'>
-				<div id="ballot" class="center">
-				<b><center>Most Preferred</center></b>
-				<ol id="sortable" class="rankings" style="padding:0px">
-			"}
-			for(var/O in options)
-				var/datum/polloption/PO = options["[O]"]
-				if(PO.optionid && PO.optiontext)
-					output += "<li voteid='[PO.optionid]' class='ranking'><span class='grippy'></span> [PO.optiontext]</li>\n"
-			output += {"
-				</ol>
-					<b><center>Least Preferred</center></b><br>
-				</div>
-					<p><input type='submit' value='[( votedfor.len ? "Re" : "")]Vote'></form>
-			"}
-			src << browse(null ,"window=playerpolllist")
-			src << browse(output,"window=playerpoll;size=500x500")
-	return
+/**
+  * Shows voting window for an IRV type poll, listing its options and relevant details.
+  *
+  * If already voted on, the options are sorted how a player voted for them, otherwise they are randomly shuffled.
+  *
+  */
+/mob/dead/new_player/proc/poll_player_irv(datum/poll_question/poll)
+	var/datum/asset/irv_assets = get_asset_datum(/datum/asset/group/IRV)
+	irv_assets.send(src)
+	var/datum/DBQuery/query_irv_get_votes = SSdbcore.NewQuery("SELECT optionid FROM [format_table_name("poll_vote")] WHERE pollid = [sanitizeSQL(poll.poll_id)] AND ckey = '[sanitizeSQL(ckey)]' AND deleted = 0")
+	if(!query_irv_get_votes.warn_execute())
+		qdel(query_irv_get_votes)
+		return
+	var/list/voted_for = list()
+	while(query_irv_get_votes.NextRow())
+		voted_for += text2num(query_irv_get_votes.item[1])
+	qdel(query_irv_get_votes)
+	var/list/prepared_options = list()
+	//if they've already voted we use the order they voted in plus a shuffle of any options they haven't voted for, if any
+	if(length(voted_for))
+		for(var/vote_id in voted_for)
+			for(var/o in poll.options)
+				var/datum/poll_option/option = o
+				if(option.option_id == vote_id)
+					prepared_options += option
+		var/list/shuffle_options = poll.options - prepared_options

```

Comment:
new code above

old code below

https://github.com/Jordie0608/tgstation/blob/cf7e8aa9ec1da95efc4c8e4e8722e7525e43fc4e/code/modules/mob/dead/new_player/poll.dm#L268

```dm
				var/list/datum/polloption/newoptions = list()
				for (var/V in votedfor)
					var/datum/polloption/PO = options["[V]"]
					if(PO)
						newoptions["[V]"] = PO
						options -= "[V]"
				//add any options that they didn't vote on (some how, some way)
				options = shuffle(options)
```

I did it this way for a reason, as i always do.

https://www.youtube.com/watch?v=FSmbsHSOEAg - reason